pkg/security/resolvers/process/resolver_windows.go

```diff
@@ -187,28 +190,50 @@ func (p *Resolver) Snapshot() {
 	// the `Cmdline` is an array of strings, parsed on ` ` boundaries
 	// the `stats` field is mostly not filled in because of the `false` argument to `ProcessesByPID()`
 	//     however, the create time will be filled in
+
+	var entries []*model.ProcessCacheEntry
+
 	for pid, proc := range pmap {
 		e := p.processCacheEntryPool.Get()
 		e.PIDContext.Pid = Pid(pid)
 		e.PPid = Pid(proc.Ppid)
 
 		e.Process.Args = strings.Join(proc.GetCmdline(), " ")
 		e.Process.FileEvent.PathnameStr = proc.Exe
-		e.Process.FileEvent.BasenameStr = path.Base(e.Process.FileEvent.PathnameStr)
-		e.ExecTime = time.Now()
+		e.Process.FileEvent.BasenameStr = filepath.Base(e.Process.FileEvent.PathnameStr)
+		e.ExecTime = time.Unix(0, proc.Stats.CreateTime)
 
-		p.insertEntry(e)
+		entries = append(entries, e)
 
 		log.Tracef("PID %d  %d PPID %d\n", pid, proc.Pid, proc.Ppid)
 		log.Tracef("  executable %s\n", proc.Exe)
 		log.Tracef("  executable %v\n", proc.GetCmdline())
 		log.Tracef("  createtime %v\n", proc.Stats.CreateTime)
+		log.Tracef("  exectime %s\n", e.ExecTime)
+
+		// TODO:
+		// another note on PPids.  Windows reuses process IDS.  So consider the following
+
+		// process 1 starts
+		// process 1 starts process 2 (so 1 is the parent of 2)
+		// process 1 ends/dies
+		// another process starts and is given the pid (1)
+		// process 2's PPid will still be 2, but the current Pid(1) was not the one that created pid 2.
 	}
-	// another note on PPids.  Windows reuses process IDS.  So consider the following
 
-	// process 1 starts
-	// process 1 starts process 2 (so 1 is the parent of 2)
-	// process 1 ends/dies
-	// another process starts and is given the pid (1)
-	// process 2's PPid will still be 2, but the current Pid(1) was not the one that created pid 2.
+	// make sure to insert them in the creation time order
+	sort.Slice(entries, func(i, j int) bool {
+		entryA := entries[i]
+		entryB := entries[j]
+
+		if entryA.ExecTime == entryB.ExecTime {
```

Comment:
I think we should use `.Equal` here, because time objects can have some TZ data and point to the same time even though they have different internal values